python/ray/ml/predictor.py

```diff
@@ -32,18 +67,58 @@ def from_checkpoint(cls, checkpoint: Checkpoint, **kwargs) -> "Predictor":
         """
         raise NotImplementedError
 
+    @DeveloperAPI
+    @abc.abstractmethod
+    def predict_pandas(self, data: pd.DataFrame, **kwargs) -> pd.DataFrame:
```

Comment:
Could we actually add leading underscore? I know it's developer API but most users shouldn't need to know about this.

Reply:
+1

Reply:
We had a discussion about this recently with @richardliaw - IIRC the consensus was that users (developers) implementing a Predictor shouldn't need to overwrite a private method. The reasoning was (IIRC) that we don't guarantee API stability for private APIs.
However, if we argue that `_` only concerns the _access scope_, I'd be fine with underscoring this.

Reply:
Ok will make this private for now. But we need to document standards on the conventions here across the entire Ray repo. There's definitely been some confusion with different teams having different conventions.